.azure-pipelines-templates/matrix.yml

```diff
@@ -1,116 +1,170 @@
       container: nosgx
       pool: Ubuntu-1804-D8s_v3
-      dependsOn: ['BuildAndTest_SGX_CFT_Perf', 'BuildAndTest_SGX_BFT_Perf']
+      dependsOn: ['Test_SGX_CFT_Perf', 'Test_SGX_BFT_Perf']
     Publish:
       container: nosgx
       pool: Ubuntu-1804-D8s_v3
-      dependsOn: ['BuildAndTest_SGX_CFT_Release', 'BuildAndTest_SGX_BFT_Release']
-    Perf:
+      dependsOn: ['Test_SGX_CFT_Release', 'Test_SGX_BFT_Release']
+    Build_Perf:
       pool: CCF-Perf
       dependsOn:  ['Formatting_and_License_Checks']
+    Perf:
+      pool: CCF-Perf
+      dependsOn:  ['Build_SGX_Perf']
 
   build:
-    common:
-      cmake_args: '-DCMAKE_C_COMPILER_LAUNCHER="ccache" -DCMAKE_CXX_COMPILER_LAUNCHER="ccache"'
-    NoSGX:
-      cmake_args: '-DTARGET=virtual -DCOVERAGE=ON'
-    SGX:
-      cmake_args: ''
-    BFT:
-      cmake_args: '-DPBFT=ON'
-    CFT:
-      cmake_args: ''
-    debug:
-      cmake_args: '-DCMAKE_BUILD_TYPE=Debug -DVERBOSE_LOGGING=ON -DBUILD_SMALLBANK=OFF'
-    install:
-      install_prefix: '/tmp/ccf-install'
-      cmake_args: '-DCMAKE_INSTALL_PREFIX=/tmp/ccf-install'
-    san:
-      cmake_args: '-DSAN=ON'
-    perf:
-      cmake_args: '-DVERBOSE_LOGGING=ON -DDISTRIBUTE_PERF_TESTS="`../.nodes.sh`"'
+    Build_SGX: '-DCMAKE_C_COMPILER_LAUNCHER="ccache" -DCMAKE_CXX_COMPILER_LAUNCHER="ccache" -DCMAKE_BUILD_TYPE=Debug -DVERBOSE_LOGGING=ON -DBUILD_SMALLBANK=OFF'
+    Build_NoSGX: '-DCMAKE_C_COMPILER_LAUNCHER="ccache" -DCMAKE_CXX_COMPILER_LAUNCHER="ccache" -DCMAKE_BUILD_TYPE=Debug -DVERBOSE_LOGGING=ON -DBUILD_SMALLBANK=OFF -DTARGET=virtual -DCOVERAGE=ON'
+    SGX: '-DCMAKE_C_COMPILER_LAUNCHER="ccache" -DCMAKE_CXX_COMPILER_LAUNCHER="ccache" -DCMAKE_BUILD_TYPE=Debug -DVERBOSE_LOGGING=ON -DBUILD_SMALLBANK=OFF'
+    NoSGX: '-DCMAKE_C_COMPILER_LAUNCHER="ccache" -DCMAKE_CXX_COMPILER_LAUNCHER="ccache" -DCMAKE_BUILD_TYPE=Debug -DVERBOSE_LOGGING=ON -DBUILD_SMALLBANK=OFF -DTARGET=virtual -DCOVERAGE=ON'
+    san: '-DCMAKE_C_COMPILER_LAUNCHER="ccache" -DCMAKE_CXX_COMPILER_LAUNCHER="ccache" -DSAN=ON -DTARGET=virtual -DCOVERAGE=ON'
+    perf: '-DCMAKE_C_COMPILER_LAUNCHER="ccache" -DCMAKE_CXX_COMPILER_LAUNCHER="ccache" -DVERBOSE_LOGGING=ON -DDISTRIBUTE_PERF_TESTS="`../.nodes.sh`"  -DCMAKE_INSTALL_PREFIX=/tmp/ccf-install'
+    install: '-DCMAKE_C_COMPILER_LAUNCHER="ccache" -DCMAKE_CXX_COMPILER_LAUNCHER="ccache" -DCMAKE_INSTALL_PREFIX=/tmp/ccf-install'
+
+  install:
+    install_prefix: '/tmp/ccf-install'
 
 
   test:
     NoSGX:
-      ctest_args: '-LE "benchmark|perf|suite"'
+      CFT:
+        ctest_args: '-LE "benchmark|perf|suite|pbft"'
+      BFT:
+        ctest_args: '-LE "benchmark|perf|suite|raft"'
     SGX:
-      ctest_args: '-LE "benchmark|perf"'
+      CFT:
+        ctest_args: '-LE "benchmark|perf|pbft"'
+      BFT:
+        ctest_args: '-LE "benchmark|perf|raft"'
     Release:
-      ctest_args: '-LE "benchmark|perf"'
+      CFT:
+        ctest_args: '-LE "benchmark|perf|pbft"'
+      BFT:
+        ctest_args: '-LE "benchmark|perf|raft"'
```

Comment:
That also does not seem right, there could just be Raft and BFT entries under test, and the command can compose those with the target flags (-LE etc).

Reply:
For the record, I was wrong about this, ctest does not process multiple flags according to GNU convention and only keeps the last value.